doc/sphinx-guides/source/installation/config.rst

```diff
@@ -3343,6 +3343,12 @@ please find all known feature flags below. Any of these flags can be activated u
     * - api-session-auth
       - Enables API authentication via session cookie (JSESSIONID). **Caution: Enabling this feature flag exposes the installation to CSRF risks!** We expect this feature flag to be temporary (only used by frontend developers, see `#9063 <https://github.com/IQSS/dataverse/issues/9063>`_) and for the feature to be removed in the future.
       - ``Off``
+    * - api-bearer-auth
```

Comment:
FWIW: I originally thought that there would be a separate api-bearer-registration flag rather than overloading this flag. Is it always true that if people allow bearer tokens that also want people to be able to create new accounts this way? I'm not sure, but unless someone thinks this use case is real, I'm OK with it as is with one flag.

Reply:
There is a separate flag. I'm not sure if perhaps you hadn't noticed it. There are two flags: `api-bearer-auth `and `api-bearer-auth-provide-missing-claims` (the one used in registration when the provider's claims are insufficient). I added both to the docs, as `api-bearer-auth` was missing from the Feature Flags section.

Reply:
Right. I meant 3 - allow authorizing by token, allow reg by token, allow reg by token with user supplied info. At QDR (a fork) we create accounts via a different mechanism tied to our single sign on mechanism and will eventually want api access via token but not registration via token. As I said above, I don't know that there's a real non-fork use case where other people will want this, so I think it is OK to stay with just the 2 flags you have. We can make a change in the future if needed.

Reply:
There is, we want account generation via token but don't allow the user to supply information.

Reply:
I think we're talking about different things. I was asking whether there's a case where you'd want tokens to work but not allow new accounts to be created with them. To address your point though: I think the api-bearer-auth-provide-missing-claims flag in the PR is already trying to address what you want, except, as you point out, the flag doesn't cover the terms in json. I think it could make sense to treat the terms accepted json the same as name/email and make the api-bearer-auth-provide-missing-claims flag cover all three, or perhaps better to have another flag (api-bearer-auth-terms-accepted-via-OIDC?) - with documentation that you must then make sure your client is configured to present the terms (and/or name/email) during the initial consent.

While I know we want to get this PR out, perhaps it makes sense to add an api-bearer-auth-terms-accepted-via-OIDC flag now and add the ~same logic we have for name/email for the terms (you can't submit the json if not allowed)? If I understand, that would allow a pure Keycloak config (except for role/affiliation which are not required fields).